pgoapi/pgoapi.py

```diff
@@ -623,39 +639,65 @@ def cleanup_pokemon(self, inventory_items=None):
         caught_pokemon = self.get_caught_pokemons(inventory_items)
         for pokemons in caught_pokemon.values():
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
-                # highest lvl pokemon first
-                sorted_pokemons = sorted(pokemons, key=self.pokemon_lvl, reverse=True)
-                for pokemon in sorted_pokemons[self.MIN_SIMILAR_POKEMON:]:
-                    if self.is_pokemon_eligible_for_transfer(pokemon, sorted_pokemons[0]):
+                # highest scoring pokemon first
+                sorted_pokemons = sorted(pokemons, key=self.pokemon_score, reverse=True)
+                pokemons_keep_counter = 0
+                pokemons_keep_iv_counter = 0
+                for pokemon in sorted_pokemons:
```

Comment:
since we really never throw away the best pokemon, why not just do sorted_pokemon[min_similar_pokemon:] then no need to check it everytime down there. And we can set pokemon_keep_counter = 1 to start with.